src/features/Students/StudentsFilters/index.jsx

```diff
@@ -1,69 +1,219 @@
+import React, {
+  useContext, useEffect, useReducer, useState,
+} from 'react';
 import { Col, Form } from '@edx/paragon';
-import React from 'react';
+import { logError } from '@edx/frontend-platform/logging';
+import { camelCaseObject } from '@edx/frontend-platform';
+import { Select, Button } from 'react-paragon-topaz';
+import { getCoursesByInstitution, getClassesByInstitution } from 'features/Students/data/api';
+import { InstitutionContext } from 'features/Main/institutionContext';
+import reducer from 'features/Students/StudentsFilters/reducer';
+import {
+  FETCH_COURSES_DATA_REQUEST,
+  FETCH_COURSES_DATA_SUCCESS,
+  FETCH_COURSES_DATA_FAILURE,
+  FETCH_CLASSES_DATA_REQUEST,
+  FETCH_CLASSES_DATA_SUCCESS,
+  FETCH_CLASSES_DATA_FAILURE,
+} from 'features/Students/actionTypes';
+import { RequestStatus } from 'features/constants';
 import PropTypes from 'prop-types';
 
-export const StudentsFilters = props => {
-  const {
-    filters,
-    setFilters,
-  } = props;
+const initialState = {
+  courses: {
+    data: [],
+    status: RequestStatus.SUCCESS,
+    error: null,
+  },
+  classes: {
+    data: [],
+    status: RequestStatus.SUCCESS,
+    error: null,
+  },
+};
+
+const StudentsFilters = ({ resetPagination, fetchData }) => {
+  const stateInstitution = useContext(InstitutionContext);
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const [courseOptions, setCourseOptions] = useState([]);
+  const [classesOptions, setClassesOptions] = useState([]);
+  const [studentName, setStudentName] = useState('');
+  const [studentEmail, setStudentEmail] = useState('');
+  const [courseSelected, setCourseSelected] = useState(null);
+  const [classSelected, setClassSelected] = useState(null);
+  const [statusSelected, setStatusSelected] = useState(null);
+  const [examSelected, setExamSelected] = useState(null);
+  // check this after implementation of selector institution
```

Comment:
How will this change?

Reply:
The information of the institution selected won't be an array, just an object so we don't have to verify the length